src/framework/ui/tab/tab.component.tsx

```diff
@@ -41,38 +39,32 @@ export class Tab extends React.Component<Props> {
         justifyContent: 'center',
         alignItems: 'center',
       },
-      icon: {},
+      icon: source.icon,
       title: source.text,
     };
   };
 
-  private createTextComponent = (style: StyleType): React.ReactElement<TextProps> => {
-    return (
+  private createTextComponent = (style: StyleType): React.ReactElement<TextProps> | null => {
+    return this.props.title ? (
       <Text
         style={style}
         key={1}>
         {this.props.title}
       </Text>
-    );
+    ) : null;
   };
 
-  private createImageComponent = (style: StyleType): React.ReactElement<ImageProps> => (
-    <Image
-      style={style}
-      key={0}
-      source={this.props.icon}
-    />
-  );
-
-  private createComponentChildren = (style: StyleType): React.ReactNode => {
-    const { icon, title } = this.props;
-
-    return [
-      icon ? this.createImageComponent(style.icon) : undefined,
-      title ? this.createTextComponent(style.title) : undefined,
-    ];
+  private createImageComponent = (style: StyleType): React.ReactElement<ImageProps> | null => {
+    const icon: React.ReactElement<ImageProps> =
+      this.props.icon && this.props.icon(style.width, style.height, style.color);
+    return icon ? React.cloneElement(icon, { key: 2 }) : null;
   };
 
+  private createComponentChildren = (style: StyleType): React.ReactNode => ([
+    this.createImageComponent(style.icon),
```

Comment:
You can check for children creation need right here.
See [example](https://github.com/akveo/react-native-ui-kitten/blob/next/src/framework/ui/tab/tab.component.tsx#L67)

Reply:
Don't see the difference.